shared/src/commonMain/kotlin/io/customer/shared/serializer/CustomAttributeContextualSerializer.kt

```diff
@@ -0,0 +1,108 @@
+package io.customer.shared.serializer
+
+import io.customer.shared.serializer.CustomAttributeContextualSerializer.Companion.DEFAULT_FALLBACK_VALUE
+import io.customer.shared.util.Logger
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.*
+
+/**
+ * Contextual serializer for Kotlin serialization to add support for generic objects. The
+ * serializer assures the parsing is safe and fallbacks to [DEFAULT_FALLBACK_VALUE] where it
+ * fails to process.
+ *
+ * To support [Any] object in the class, the class file should add the following annotation
+ *
+ * [@file:UseContextualSerialization(Any::class)]
+ */
+internal class CustomAttributeContextualSerializer(
```

Comment:
This custom serializer has worked well for me so far. But I would like to keep eye on it and try sending maximum types in `CustomAttributes` and see if they work as expected or not.

Reply:
I am new to the Kotlin serialization library. I was expecting this class to be much smaller and have Kotlin serialization do more of the logic? 

Can you help explain why this code is needed?